x/evm/statedb/state_object.go

```diff
@@ -0,0 +1,245 @@
+package statedb
+
+import (
+	"bytes"
+	"math/big"
+	"sort"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+)
+
+var emptyCodeHash = crypto.Keccak256(nil)
+
+// Account is the Ethereum consensus representation of accounts.
+// These objects are stored in the main account trie.
+type Account struct {
```

Comment:
thoughts about moving this type to `types/`? cc @tomtau

Reply:
to my understanding, `types/` contain the external facing types... right now, they have the proto account type.
I think it'd perhaps make sense to move if e.g. in Web3 API, the same account struct was being returned somewhere / converted from the proto one, so it could be de-duplicated.
I only found:
https://github.com/tharsis/ethermint/blob/514785bd89b7a6cd75327762c316ae844124135c/rpc/ethereum/namespaces/eth/api.go#L924 and AccountResult in RPC is a superset... perhaps this statedb's account can be embedded in it, but I guess it'd require custom marshalling and unmarshalling functions that would flatten the fields.

Reply:
keep it here can keep statedb package self-contained, currently, it doesn't rely on the other part of ethermint.